### DISCUSSION TITLE
How to access old record data after save?

### DISCUSSION THREAD

**Discussion by afbora** [filamentphp/filament#11579] [Help] [answered]
### Package

Panel builder

### Package Version

3.2.0

### How can we help you?

I'm trying to do compare old and new record to send notification to the user. For example, send notification if the order status has changed. But I could not find in the documentation how to access the old data after the model is saved.

````php
protected function afterSave(): void
{
    if ($oldRecordStatus !== $this->record->status) {
        Notification::make()->send();
    }
}
````

**Comment by leandrocfe** [comment, marked as answer]
Maybe using [Observers](https://laravel.com/docs/10.x/eloquent#observers)?

I think you can use `isDirty()`
```php
public function updating(YourModel $yourModel)
{
    if($yourModel->isDirty('status')) {
    //...
    }
}
```

**Comment by afbora** [comment]
Works like a charm! Thank you. 
I didn't know about `isDirty()` and `wasChanged()` methods. `wasChanged` works perfect for me.

````php
protected function afterSave(): void
{
    if ($this->record->wasChanged('status')) {
        if ($user = User::find($this->record->getAttribute('user_id'))) {
            Notification::make()
                ->success()
                ->sendToDatabase($user);
        }
    }
}
````
